packages/flutter/lib/src/painting/_network_image_io.dart

```diff
@@ -64,22 +65,16 @@ class NetworkImage extends image_provider.ImageProvider<image_provider.NetworkIm
   // We set `autoUncompress` to false to ensure that we can trust the value of
   // the `Content-Length` HTTP header. We automatically uncompress the content
   // in our call to [consolidateHttpClientResponseBytes].
-  //
-  // In order to allow HttpOverrides to be used in different parts of a Flutter
-  // application, use these or the parent zone as a key to look up the client.
-  static final Expando<HttpClient> _sharedClients = Expando<HttpClient>();
-  static HttpClient get _sharedHttpClient {
-    final Object token = HttpOverrides.current ?? Zone.root;
-    HttpClient? cachedClient = _sharedClients[token];
-    cachedClient ??= (_sharedClients[token] = HttpClient()..autoUncompress = false);
-    return cachedClient;
-  }
+  static final HttpClient _sharedHttpClient = HttpClient()..autoUncompress = false;
 
   static HttpClient get _httpClient {
     HttpClient client = _sharedHttpClient;
     assert(() {
       if (debugNetworkImageHttpClientProvider != null)
         client = debugNetworkImageHttpClientProvider!();
+     // do not use the cached client if an HttpOverrides is defined for testing.
+      else if (HttpOverrides.current != null)
+        client = HttpClient()..autoUncompress = false;
```

Comment:
This seems like it'll be strange because tests and debug applications will pass, but profile/release applications will not.

If we have a publicly, non-debug-only way to set this, it should just set it. I wonder if we should just kill off `debugNetworkImageHttpClientProvider` with a deprecation notice to just use `HttpOverrides`.

I'm also not clear on the implications of changing this property out from under people. I was thinking this would be something like 

```
  client = HttpOverrides.current?.createHttpClient() ?? _sharedHttpClient;
  assert(() {
    if (client.autoUncompress) {
      // Warn developers that we're going to flip this?
    }
    return true;
  }());
  client.autoUncompress = false;
```

I'm not really sure about the right way to handle that warning. There may be no action to take, since they normally want it true but we need it false. So maybe we should just set it to false and not worry abou tit.

Reply:
Well practically, I don't think there are any real implementations of the HttpClient class besides the dart:io one. So at most I could see these used for testing with a fake, or for configuring the existing client in a slightly different way. I'm not really too worried about people doing weird stuff with their client, it will break and they will have to figure it out - but they had to go out of their way to use an HttpOverrides anyway.

The advantage of `debugNetworkImageHttpClientProvider ` is that you don't need to muck around with zones. It might predate `HttpOverrides`, but I don't quite remember

Reply:
If you use `HttpOverrides.global` the zone mucking is largely obscured too though.

I'm still not clear on why we would ignore this in a mobile or desktop based app that sets it in `main`, e.g. to make sure that the http client always uses a particular security context for a network certificate...

Reply:
We could definitely make it not debug only, but then we need to go back to caching - otherwise each network image run in an httpoverrides would create its own client - which we're trying to avoid